JamfSync/Multipart/AmazonS3.swift

```diff
@@ -0,0 +1,340 @@
+//
+//  AmazonS3.swift
+//
+
+import Cocoa
+import CryptoKit
+import Foundation
+
+
+class CompleteMultipart: NSObject {
+    var partNumber: Int
+    var eTag: String
+    
+    init(partNumber: Int, eTag: String) {
+        self.partNumber = partNumber
+        self.eTag = eTag
+    }
+}
+
+struct Chunk {
+    static var all               = 0
+    static var size              = 1024*1024*10       //  1024*1024 = 1 MB
+    static var index             = 1
+    static var numberOf          = 0
+    static var previousSignature = ""
+    static var authHeader        = ""
+}
+
+// for aws uri encoding
+//var myCharacterSet = CharacterSet()
+var partNumberEtagList = [CompleteMultipart]()
+
+func tagValue(xmlString:String, startTag:String, endTag:String, includeTags: Bool) -> String {
+    var rawValue = ""
+    if let start = xmlString.range(of: startTag),
+        let end  = xmlString.range(of: endTag, range: start.upperBound..<xmlString.endIndex) {
+        rawValue.append(String(xmlString[start.upperBound..<end.lowerBound]))
+    }
+    if includeTags {
+        return "\(startTag)\(rawValue)\(endTag)"
+    } else {
+        return rawValue
+    }
+}
+
+/*
+class AmazonS3: NSObject, URLSessionDelegate {
+    
+    static let shared = AmazonS3()
+    private override init() {
```

Comment:
Probably best to remove all the commented out code before we continue with the pull request review. That code will still be in the commit history, we could refer to that if we need to look at anything you commented out. This whole init function can be removed since it would be empty after commenting out the code.